brewtils/rest/client.py

```diff
@@ -442,6 +443,31 @@ def get_user(self, user_identifier):
         """
         return self.session.get(self.user_url + user_identifier)
 
+    @enable_auth
+    def get_file(self, file_id, **kwargs):
+        return self.session.get(self.file_url + file_id, **kwargs)
+
+    @enable_auth
+    def post_files(self, files):
+        """Performs a POST on the File URL.
+
+        files should be in the form:
+
+            {"identifier": ("desired_filename", open("filename", "rb")) }
+
+        :param files: Dictionary of filename to files.
+        :return: Response to the request
+        """
+
+        # This is here in case we have not authenticated yet. Without this
+        # code, it is possible for us to perform the POST, which will call
+        # read on each of the files, that method fails with a 401, we
+        # then authenticate and try again, only to post an empty file.
+        for info in files.values():
+            info[1].seek(0)
```

Comment:
Would it make more sense to do some sort of preflight-type thing here to kick the auth mechanism? It seems like we're sending the files twice in this case which is a potentially large performance hit.

Reply:
The only thing I can think of here is a call to get your tokens as a pre-flight check. Other than that, we should just replicate the enable_auth function